python/__init__.py

```diff
@@ -624,14 +625,17 @@ def set_can_speed_kbps(self, bus, speed):
   def set_can_data_speed_kbps(self, bus, speed):
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xf9, bus, int(speed * 10), b'')
 
-  # CAN FD and BRS status
+  # CAN FD, BRS, non-ISO mode status
   def get_canfd_status(self, bus):
-    dat = self._handle.controlRead(Panda.REQUEST_IN, 0xfa, bus, 0, 2)
+    dat = self._handle.controlRead(Panda.REQUEST_IN, 0xfa, bus, 0, 3)
     if dat:
```

Comment:
Would add length check too as python lib will get exception from struct.unpack if old firmware panda will be connected.

Reply:
sure, can pad with zero byte and it will be backwards compatible (was always non-ISO in the past)

Reply:
I don't think backwards compatibility is a good goal here; it introduces a lot of complexity: when should we care about backwards compatibility? how long should I keep this code around? etc. I would just add `ensure_can_packet_version` to this function.

Reply:
Does tying it to `ensure_can_packet_version` make sense?  That seems confusing, too, since it has no impact on what is being returned here (we didn't change that version for this PR).

Adding a new version number isn't trivial either.  If we care that much about adding `.ljust(3, b"\x00")` here I would recommend just making it a breaking change; no one is using this method anyway.

Reply:
we could also make it a new method with a new ID, but I didn't think it was worth the extra lines of code required.

Reply:
Meh, then let's just break it.

Reply:
I feel strongly that priorities should be: communicating why something doesn't work > doesn't work (uncommunicated) > half works because we added some backwards compatibility. In this case, I'd probably just remove `get_canfd_status`; isn't it now just a subset of the can health stuff?

Reply:
Looks like it is a subset so it could be removed, but if we remove it any old scripts using it get `doesn't work (uncommunicated)` and if we instead add a length check as originally suggested you get `communicating why something doesn't work` so should we keep it and add a length check?

Reply:
realistically, no one is using this in a script. removing is the best option in this case